pyproject.toml

```diff
@@ -1,94 +1,110 @@
-
-[tool.poetry] # https://python-poetry.org/docs/pyproject/
+[project] # https://packaging.python.org/en/latest/specifications/pyproject-toml/
 name = "raglite"
 version = "0.6.2"
 description = "A Python toolkit for Retrieval-Augmented Generation (RAG) with SQLite or PostgreSQL."
-authors = ["Laurent Sorber <laurent@superlinear.eu>"]
 readme = "README.md"
-repository = "https://github.com/superlinear-ai/raglite"
-
-[tool.commitizen] # https://commitizen-tools.github.io/commitizen/config/
+authors = [
+  { name = "Laurent Sorber", email = "laurent@superlinear.eu" },
+]
+requires-python = ">=3.10,<4.0"
+dependencies = [
+  # Python:
+  # "python (>=3.10,<4.0)",
+  # Configuration:
+  "platformdirs (>=4.0.0)",
+  # Markdown conversion:
+  "pdftext (>=0.3.13)",
+  "scikit-learn (>=1.4.2)",
+  # Markdown formatting:
+  "markdown-it-py (>=3.0.0)",
+  "mdformat-gfm (>=0.3.6)",
+  # Sentence and chunk splitting:
+  "numpy (>=1.26.4,<2.0.0)",
+  "scipy (>=1.5.0)",
+  "wtpsplit-lite (>=0.1.0)",
+  # Large Language Models:
+  "huggingface-hub (>=0.22.0)",
+  "litellm (>=1.48.4,<1.56.10)",
+  "llama-cpp-python (>=0.3.2)",
+  "pydantic (>=2.7.0)",
+  # Approximate Nearest Neighbors:
+  "pynndescent (>=0.5.12)",
+  "numba (>=0.60.0)",
+  # Reranking:
+  "langdetect (>=1.0.9)",
+  # Storage:
+  "pg8000 (>=1.31.2)",
+  "sqlmodel-slim (>=0.0.18)",
+  # Progress:
+  "tqdm (>=4.66.0)",
+  # Evaluation:
+  "pandas (>=2.1.0)",
+  # CLI:
+  "typer (>=0.15.1)",
+  # Model Context Protocol:
+  "fastmcp (>=0.4.1)",
+  # Utilities:
+  "packaging (>=23.0)",
+]
+
+[project.scripts]  # https://docs.astral.sh/uv/concepts/projects/config/#command-line-interfaces
+raglite = "raglite.cli:app"
+
+[project.urls]  # https://packaging.python.org/en/latest/specifications/well-known-project-urls/#well-known-labels
+homepage = "https://github.com/superlinear-ai/raglite"
+source = "https://github.com/superlinear-ai/raglite"
+changelog = "https://github.com/superlinear-ai/raglite/blob/main/CHANGELOG.md"
+releasenotes = "https://github.com/superlinear-ai/raglite/releases"
+documentation = "https://github.com/superlinear-ai/raglite"
+issues = "https://github.com/superlinear-ai/raglite/issues"
+
+[dependency-groups]  # https://docs.astral.sh/uv/concepts/projects/dependencies/#development-dependencies
+test = [
+  "commitizen (>=3.29.1)",
+  "coverage (>=7.6.1)",
+  "mypy (>=1.9.0)",
+  "poethepoet (>=0.25.0)",
+  "pre-commit (>=3.7.0)",
+  "pytest (>=8.1.1)",
+  "pytest-mock (>=3.14.0)",
+  "ruff (>=0.5.7)",
+  "safety (>=3.1.0)",
+  "shellcheck-py (>=0.10.0.1)",
+  "typeguard (>=4.2.1)",
+]
+dev = [
+  "commitizen (>=4.1.0)",
+  "coverage[toml] (>=7.6.10)",
+  "ipykernel (>=6.29.4)",
+  "ipython (>=8.18.0)",
+  "ipywidgets (>=8.1.2)",
+  "mypy (>=1.14.1)",
+  "pdoc (>=15.0.1)",
+  "poethepoet (>=0.32.1)",
+  "pre-commit (>=4.0.1)",
+  "pytest (>=8.3.4)",
+  "pytest-mock (>=3.14.0)",
+  "pytest-xdist (>=3.6.1)",
+  "ruff (>=0.9.2)",
+  "typeguard (>=4.4.1)",
+]
```

Comment:
I would merge these under a single `dev` group since we don't differentiate between these groups anywhere. (It's true that we could differentiate in CI by only installing the run time and test dependencies, but we don't currently do that.)